### ISSUE TITLE
Traktor Kontrol Z1 - RangeError: index out of range

### ISSUE THREAD

**Issue by JoergAtGithub** [mixxxdj/mixxx#13832]
I´m a new user and when I tried to connect my unity to mix, and error was reported. Could anyone help me, please?:

Uncaught exception: file:///Volumes/mixxx-2.4.1/Mixxx.app/Contents/Resources/controllers/common-hid-packet-parser.js:455: RangeError: index out of range
Backtrace: unpack@file:///Volumes/mixxx-2.4.1/Mixxx.app/Contents/Resources/controllers/common-hid-packet-parser.js:455
parse@file:///Volumes/mixxx-2.4.1/Mixxx.app/Contents/Resources/controllers/common-hid-packet-parser.js:989
parsePacket@file:///Volumes/mixxx-2.4.1/Mixxx.app/Contents/Resources/controllers/common-hid-packet-parser.js:1636
readCurrentPosition@file:///Volumes/mixxx-2.4.1/Mixxx.app/Contents/Resources/controllers/Traktor-Kontrol-Z1-scripts.js:123
init@file:///Volumes/mixxx-2.4.1/Mixxx.app/Contents/Resources/controllers/Traktor-Kontrol-Z1-scripts.js:26

_Originally posted by @Karaku in https://github.com/mixxxdj/mixxx/issues/1989#issuecomment-2453523144_

**Comment by Karaku** [comment]
Hey! Is working! 


After modifying the code, the option to map it via midi with the learning assistant via midi has been enabled. I still need to map the Main Knob and the FX "On" and "Mode" buttons, if someone would be so kind as to Explain to me if they have any correspondence, thanks again!

The other two options have these errors:

The mapping "Apple Mikey HID Driver " and "Traktor Kontrol Z1 F76A_3":

Uncaught exception: file:///Applications/Mixxx.app/Contents/Resources/controllers/Traktor-Kontrol-Z1-scripts.js:129: SyntaxError: Unexpected token `identifier'
Backtrace: @file:///Applications/Mixxx.app/Contents/Resources/controllers/Traktor-Kontrol-Z1-scripts.js:129

**Comment by JoergAtGithub** [comment]
The mapping that comes with Mixxx is a HID mapping, you must load it to the Traktor Z1 **HID** device, not to the MIDI device (which is just a software emulation).

**Comment by Karaku** [comment]
Sorry, I'm quite new to these issues, I see that in the Controllers folder, in addition to the .js file, there is a hid.txt file and a hid.txt file, but line 119 does not exist there... :)

**Comment by JoergAtGithub** [comment]
The file name on your system is: file:///Applications/Mixxx.app/Contents/Resources/controllers/Traktor-Kontrol-Z1-scripts.js

**Comment by Karaku** [comment]
Yes, that is the file that I modified with line 119 and as a result it only let me map the controller manually...

**Comment by JoergAtGithub** [comment]
Do you get any error message after the change? Can you share a Screenshot of your controller settings in the Preference dialog?

**Comment by JoergAtGithub** [comment]
@djantti Do you see the same issue with the Z1 mapping as reported here? Does the path suggested above work for you: https://github.com/mixxxdj/mixxx/issues/13832#issuecomment-2453564985

**Comment by djantti** [comment]
Nope, the Z1 mapping has been working perfectly on all minor and major Mixxx versions since 2.4.0. I'm running Linux, but I did try my Z1 and Mixxx 2.5.0 on a Windows VM with USB passthrough. No issues there either.

**Comment by JoergAtGithub** [comment]
Thanks for testing!
